GitExtensions/Program.cs

```diff
@@ -320,6 +317,22 @@ private static bool LocateMissingGit()
             }
         }
 
+        private static void UncaughtExceptionHandler(Exception ex)
+        {
+            if (ex is GitCommands.Config.GitConfigurationException)
+            {
+                var configEx = (GitCommands.Config.GitConfigurationException)ex;
+                string message = string.Format(Strings.GeneralGitConfigExceptionMessage, configEx.ConfigPath, Environment.NewLine, configEx.InnerException.Message);
+                MessageBox.Show(message, Strings.GeneralGitConfigExceptionCaption, MessageBoxButtons.OK, MessageBoxIcon.Error);
+                Environment.Exit(-1);
```

Comment:
I don't think we should crash the app if a submodules config is incorrect.
We could close the repo, but closing the app is a bad UX.

Reply:
@RussKie 

With so much of GE's functionality relying on the config parser, I think it is a good idea to close the repo (at this layer, I do not think there is a ).  I was not aware that we could simply route back to the "home page/repos page" instead of quitting, however.  Is there a simple command to get us back there, sort of wiping away the working memory of the current GE session as if it was a clean re-launch of the app?

Reply:
Under [Repository] menu, there is a menu item "Close (go to Dashboard)", I think we should be able to do this.
https://github.com/gitextensions/gitextensions/blob/1219252182ff0ff895cbc99c565fd5189fafd3d8/GitUI/CommandsDialogs/FormBrowse.cs#L1769-L1772

Reply:
Is going to the dashboard really better than exiting completely?

We could add a new icon for this state
(Similar to missing submodule, just a "pen" same as modified right now.)
Maybe a separate PR

Reply:
In my opinion it is, if a repo is corrupt, why do we need to kill the app? Close the repo.
In fact, for submodules config I don't think it is necessary to close the repo, tell the user everytime the config is read that it is corrupt and let the user to fix it. I don't think we should block the user from browsing commits.